pkg/cloudprovider/cloudprovider.go

```diff
@@ -165,6 +169,50 @@ func (c *CloudProvider) Delete(ctx context.Context, node *v1.Node) error {
 	return c.instanceProvider.Terminate(ctx, node)
 }
 
+func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *corev1alpha1.Machine) (bool, error) {
+	// Not needed when GetInstanceTypes removes provisioner dependency
+	provisioner := &v1alpha5.Provisioner{}
+	if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: machine.Labels[v1alpha5.ProvisionerNameLabelKey]}, provisioner); err != nil {
+		return false, fmt.Errorf("getting provisioner, %w", err)
```

Comment:
Do we need an IgnoreNotFound here?

Reply:
This usage of IgnoreNotFound introduces a bug. If the provisioner is not found we'll continue as if nothing happened. 
```suggestion
		return false, k8sClient.IgnoreNotFound(err)
```

Thinking further on this, I don't know if we want to error, since this is likely unrecoverable. Perhaps we should just print an error log and return false (not drifted).